apps/spruce/src/components/Settings/EventLog/EventDiffTable/utils.ts

```diff
@@ -0,0 +1,80 @@
+import { diff } from "deep-object-diff";
+import { isObject, omitTypename, getObjectValueByPath } from "utils/object";
+
+import { JSONObject, JSONValue } from "utils/object/types";
+import { EventDiffLine } from "../types";
+
+/**
+ * `getEventDiffLines` is a utility function that returns an array of objects representing the differences between two objects.
+ * @param before - The object before the event.
+ * @param after - The object after the event.
+ * @returns - An array of objects representing the differences between two objects.
+ */
+const getEventDiffLines = (
+  before: JSONObject | undefined | null,
+  after: JSONObject | undefined | null,
+): EventDiffLine[] => {
+  const beforeNoTypename = omitTypename(before);
+  const afterNoTypename = omitTypename(after);
+
+  const eventDiff = diff(beforeNoTypename || {}, afterNoTypename || {});
+  const pathKeys: string[] = getChangedPaths(eventDiff);
+  const eventDiffLines = pathKeys.map((key) => {
+    let previousValue = beforeNoTypename;
+    if (beforeNoTypename !== null && beforeNoTypename !== undefined) {
+      previousValue = getObjectValueByPath(beforeNoTypename, key) as JSONObject;
+    }
+    const changedValue = getObjectValueByPath(eventDiff as JSONValue, key);
+
+    const formattedKey = formatArrayElements(key);
```

Comment:
I kinda didn't test this at all besides running the corresponding test file; if it doesn't work, we should modify the test file to include a case where this will fail, because I think someone would reasonably think it could be rewritten like this
```suggestion
  const beforeNoTypename = omitTypename(before) || {};
  const afterNoTypename = omitTypename(after) || {};

  const eventDiff = diff(beforeNoTypename, afterNoTypename) as JSONValue;
  const pathKeys: string[] = getChangedPaths(eventDiff);
  const eventDiffLines = pathKeys.map((key) => {
    const previousValue = getObjectValueByPath(beforeNoTypename, key);
    const changedValue = getObjectValueByPath(eventDiff, key);
    const formattedKey = formatArrayElements(key);

```

Reply:
SGTM it worked!